daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/processors/TypeCalculator.scala

```diff
@@ -17,288 +17,9 @@
- * aribitrary subset of: minInclusive, minExclusive, maxInclusive, and maxExclusive
- */
-class RepValueSet(
-  val valueSet: HashSet[DataValuePrimitive],
-  val valueRanges: Set[(RangeBound, RangeBound)],
-) extends Serializable {
-
-  def merge(other: RepValueSet): RepValueSet = {
-    val valueSet_ = valueSet ++ other.valueSet
-    val valueRanges_ = valueRanges ++ other.valueRanges
-    new RepValueSet(valueSet_, valueRanges_)
-  }
-
-  lazy val isEmpty: Boolean = valueSet.isEmpty && valueRanges.isEmpty
-
-}
-
-//TODO, many of the key/values we receive will be BigInt.
-//We should check if we can safely convert them to Long
-
-object RepValueSetCompiler {
-  def compile(
-    valueSet: Seq[DataValuePrimitive],
-    valuesRanges: Seq[(RangeBound, RangeBound)],
-  ): RepValueSet = {
-    val hashSet = HashSet.empty ++ valueSet
-    val rangeSet = Set.empty ++ valuesRanges.filter(x => x._1.isDefined || x._2.isDefined)
-    new RepValueSet(hashSet, rangeSet)
-  }
-  def empty: RepValueSet = compile(Seq.empty, Seq.empty)
-}
-
-object TypeCalculatorCompiler {
-
-  // mappings: [(keySet, canonicalKey, value)]
-  def compileKeysetValue(
-    mappings: Seq[(RepValueSet, DataValuePrimitive, DataValuePrimitive)],
-    srcType: NodeInfo.Kind,
-    dstType: NodeInfo.Kind,
-  ): TypeCalculator = {
-    Assert.invariant(!mappings.isEmpty)
-
-    /*
-     * We need to cast to HashMap, because the type of HashMap.++ returns a generic Map
-     * HashMap.+ returns a HashMap, so we can avoid the case by doing the fold ourself
-     */
-    val valueMap: HashMap[DataValuePrimitive, DataValuePrimitive] =
-      (HashMap.empty ++ mappings.flatMap(x => {
-        val (keySet, _, value) = x
-        keySet.valueSet.map((_, value))
-      })).asInstanceOf[HashMap[DataValuePrimitive, DataValuePrimitive]]
-    val rangeTable: Seq[(RangeBound, RangeBound, DataValuePrimitive)] = mappings.flatMap(x => {
-      val (keySet, _, value) = x
-      keySet.valueRanges.map({ case (min, max) => (min, max, value) })
-    })
-    val unparseMap: HashMap[DataValuePrimitive, DataValuePrimitive] =
-      (HashMap.empty ++ mappings.map(x => {
-        val (_, canonicalKey, value) = x
-        (value, canonicalKey)
-      })).asInstanceOf[HashMap[DataValuePrimitive, DataValuePrimitive]]
-
-    /*
-     * Type erasure makes dispatching based on if we have an ordered keyset or not difficult
-     * Really, the problem is that we should not be using AnyRef at all,
-     * but since the DPath library is based around AnyRef, we are stuck with it
-     */
-    rangeTable match {
-      case Seq() =>
-        new KeysetValueTypeCalculatorUnordered(valueMap, unparseMap, srcType, dstType)
-      case _ => {
-        new KeysetValueTypeCalculatorOrdered(valueMap, rangeTable, unparseMap, srcType, dstType)
-      }
-    }
-  }
-
-  def compileIdentity(srcType: NodeInfo.Kind): TypeCalculator = new IdentityTypeCalculator(
-    srcType,
-  )
-
-}
 
 object Range {
```

Comment:
Should we rename TypeCalculator.scala to Range.scala or RangeBound.scala since it defines only those types now?

Reply:
Makes sense. Will do.